source/solvers/fluid_dynamics_matrix_free.cc

```diff
@@ -2390,22 +2395,17 @@ FluidDynamicsMatrixFree<dim>::set_initial_condition_fd(
       this->set_nodal_values();
       this->present_solution.update_ghost_values();
 
-      // Create a pointer to the current viscosity model
-      std::shared_ptr<RheologicalModel> viscosity_model =
-        this->simulation_parameters.physical_properties_manager.get_rheology();
-
-      // Keep in memory the initial viscosity
-      const double viscosity_end = viscosity_model->get_kinematic_viscosity();
+      // Get viscosity model
+      std::shared_ptr<RheologicalModel> original_viscosity_model =
+        this->system_operator->properties_manager->get_rheology();
 
-      // Set it to the initial condition viscosity
-      viscosity_model->set_kinematic_viscosity(
-        this->simulation_parameters.initial_condition->kinematic_viscosity);
+      // Temporarily set the rheology to be newtonian with predefined viscosity
+      std::shared_ptr<Newtonian> temporary_rheology =
+        std::make_shared<Newtonian>(
+          this->simulation_parameters.initial_condition->kinematic_viscosity);
 
-      // Set the kinematic viscosity in the system operator to be the
-      // temporary viscosity
-      this->system_operator->set_kinematic_viscosity(
-        this->simulation_parameters.physical_properties_manager
-          .get_kinematic_viscosity_scale());
+      this->system_operator->properties_manager->set_rheology(
+        temporary_rheology);
```

Comment:
Why not set just the property manager of the fluid dynamics class, I'm not sure i understand why you are accessing the one in the operator to do that?
See below for more discussion